### DISCUSSION TITLE
Welcome to Raven Discussions!

### DISCUSSION THREAD

**Discussion by nikkothari22** [The-Commit-Company/raven#575] [Announcements]
## 👋 Welcome!

  We’re using Discussions as a place to connect with people who are using Raven or want to contribute to it. We hope that you:
  * Ask questions you’re wondering about 
  * Share ideas or feature requests
  * Engage with other community members.

**Comment by knix2** [comment]
Will Raven be integrated with GamePlan ?

**Reply by nikkothari22** [reply]
We're currently building an integration layer that would allow you to integrate any Frappe app with Raven.

But as of now, we don't have plans for a specific direct integration with Gameplan. 

What do you have in mind?

**Reply by huembw** [reply]
hi @nikkothari22 I have an idea that user can create a task directly from a Raven msg. By that way, creating a task is convenient and that task has a clear context.
![Uploading image.png…]()

Another idea is that Gameplan users can share task/discussion/page on Raven channel, like the way they send a raven msg from ERPN. From Raven, user can quick take some actions with that object, such as change status of a task, etc,...

**Reply by nikkothari22** [reply]
Hey @huembw 

I've also been thinking about creating Tasks (ToDos) and Events directly from Raven. This is something that we'll definitely take up sometime.

**Comment by RJPvT** [comment]
Hi!

I'm using v14 frappe and erpnext and have installed as said in the readme, since 2 weeks i'm getting a node error (version mismatch) and an error : 1054 unknown replied_message_details column in field list. 

Should i be using main, develop or integration for v14 ? 
Do i use bench to switch ?

**Reply by nikkothari22** [reply]
You need to have Node v16+ on your server and you need to run bench update and make sure that it runs without error.

The "main" branch is stable whereas the "develop" branch has the latest changes. Either would work.

**Reply by RJPvT** [reply]
I have node v16: The error says so : error vite@5.1.1: The engine "node" is incompatible with this module. Expected version "^18.0.0 || >=20.0.0". Got "16.20.2"

**Comment by idanishgd** [comment]
_**NEED HELP**_

Hi,

**server**
I have installed Raven on my server its not showing online users and also after sending message i have to refresh page to know if i have received any message.

**server**
For local  its working perfectly fine.

i am using **main branch** and have also tried **develop branch** but no luck.



![Screenshot from 2024-03-27 15-05-06](https://github.com/The-Commit-Company/Raven/assets/110451173/88a8e8da-d950-47e3-85c2-49d5434e89c2)

**Reply by nikkothari22** [reply]
@danish21x Can you check if socketIO is running on your server? Raven depends on socket events to notify the client of any new messages.

To check if SocketIO is running, you can do the following:
1. Open the inspector with Raven running in the background. Go to the "Console" tab (2nd tab) and see if there are any errors.
2. On your server, check supervisor's status. `sudo supervisorctl status` - run this on the server

About the online indicator - it's a little flaky on the web app - we've noticed it as well. On the mobile app, it is able to change the status very well.

**Comment by ghost** [comment]
so when and what i can contribute to raven issue for hackathon @nikkothari22 ?

**Reply by nikkothari22** [reply]
I think the PRs only count if it's made during the days of the hackathon.

You can spend some time now to set up the repository, have a look around the codebase and try making some changes so that during the hackathon you know what to do. 

What can you contribute to for the Hackathon: issues tagged with "FOSSHack" 

More guidelines about the Hackathon here: https://fossunited.org/fosshack/2024 . Details related to the Hackathon can be provided by the FOSSHack team. We can help you with questions related to Raven.
